### PR TITLE
Redeem functionality

### PR DESCRIPTION
**Pull request by harshvitra**
Added Redeem functionality in the User profile page

- [x] Translations

### CODE REVIEW

**Comment by norbertschuler** [comment]
Is this replacing #437?

**Comment by harshvitra** [comment]
> Is this replacing #437?

No those are Redeem URLs, this is a modal in which user's can enter their codes

File: src/features/user/UserProfile/components/RedeemModal.tsx [hunk 1]
```diff
@@ -27,15 +149,131 @@ export default function RedeemModal({
...
+                      {t('me:redeem')}
+                    </h4>
+
+                    <div className={styles.inputField}>
+                      <MaterialTextField inputRef={register({
+                        required: {
+                          value: true,
+                          message: 'Please enter a code to redeem.',
+                        }
+                      })} name={'code'} placeholder="pp.eco/XADSA-DS-AS" label="" variant="outlined" />
```
Comment (norbertschuler) [inline src/features/user/UserProfile/components/RedeemModal.tsx:254]: Maybe we should change the placeholder `pp.eco/XADSA-DS-AS` to just `XADSA-DS-AS` or something else?

Comment (harshvitra) [reply src/features/user/UserProfile/components/RedeemModal.tsx:240]: we added this as the chocolates will have the link with pp.eco 

@sagararyal

**Review by norbertschuler — APPROVED** [review]

File: src/features/user/UserProfile/components/RedeemModal.tsx [hunk 2]
```diff
@@ -27,15 +143,121 @@ export default function RedeemModal({
...
+              </Snackbar>
+
+            </div>
+          </>
+        ) : (
+            <div className={styles.modal}>
+              {codeValidated && validCodeData ? (
+                <>
+                  <div className={styles.codeTreeCount}>
+                    {validCodeData.treeCount}
```
Comment (norbertschuler) [inline src/features/user/UserProfile/components/RedeemModal.tsx:213]: Format this number once this branch and https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/526 is merged!

Comment (norbertschuler) [reply src/features/user/UserProfile/components/RedeemModal.tsx:213]: This is just a reminder for myself at https://github.com/Plant-for-the-Planet-org/planet-webapp/issues/549

**Pull request merged**